src/test/librbd/exclusive_lock/test_mock_PreReleaseRequest.cc

```diff
@@ -100,11 +100,19 @@ class TestMockExclusiveLockPreReleaseRequest : public TestMockFixture {
                                bool init_shutdown, int r) {
     expect_test_features(mock_image_ctx, RBD_FEATURE_EXCLUSIVE_LOCK, true);
     if (!mock_image_ctx.clone_copy_on_read) {
-      expect_test_features(mock_image_ctx, RBD_FEATURE_JOURNALING,
-                           ((mock_image_ctx.features & RBD_FEATURE_JOURNALING) != 0));
+      if (mock_image_ctx.features & RBD_FEATURE_JOURNALING) {
+	expect_test_features(mock_image_ctx, RBD_FEATURE_JOURNALING,
+			    ((mock_image_ctx.features & RBD_FEATURE_JOURNALING) != 0));
+      } else {
+	expect_test_features(mock_image_ctx, RBD_FEATURE_JOURNALING,
+			    ((mock_image_ctx.features & RBD_FEATURE_JOURNALING) != 0));
+	expect_test_features(mock_image_ctx, RBD_FEATURE_DIRTY_CACHE,
+			    ((mock_image_ctx.features & RBD_FEATURE_DIRTY_CACHE) != 0));
+      }
```

Comment:
nit: is it github showing indentation wrong here or is it wrong indeed?

Also, it looks like the code would look a bit nicer if you did `expect_test_features(...RBD_FEATURE_JOURNALING)` unconditionally as it had been before and just added:
```
if ((mock_image_ctx.features & RBD_FEATURE_JOURNALING) == 0) {
  expect_test_features(...RBD_FEATURE_DIRTY_CACHE);
}
```

Reply:
Thanks, update by your suggestion.